File: 프로그래머스/lv2/84512. 모음 사전/모음 사전.java
class Solution {
    public int solution(String word) {
        int answer = 0;
        int[] nums = {781, 156, 31, 6, 1};
        String vowels = "AEIOU";

        for (int i = 0; i < word.length(); i++) {
            char c = word.charAt(i);
            answer += vowels.indexOf(c) * nums[i] + 1;
        }

        return answer;
    }
}

File: 프로그래머스/lv2/84512. 모음 사전/모음 사전.py
ans = 0
count = 0

def solution(word):
    alpa = 'AEIOU'
    dfs(word, '', alpa)
    return ans

def dfs(word, currentWord, alpa):
    global ans
    global count
    if currentWord == word:
        ans = count
        return
    if len(currentWord) == 5 or ans != 0:
        return

    for i in range(len(alpa)):
        count += 1
        currentWord = currentWord + alpa[i]
        dfs(word, currentWord, alpa)
        currentWord = currentWord[:len(currentWord) - 1]